source/extensions/compression/zstd/common/base.cc

```diff
@@ -0,0 +1,39 @@
+#include "source/extensions/compression/zstd/common/base.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Compression {
+namespace Zstd {
+namespace Common {
+
+Base::Base(const uint32_t chunk_size)
+    : chunk_ptr_{std::make_unique<uint8_t[]>(chunk_size)}, output_{chunk_ptr_.get(), chunk_size, 0},
+      input_{nullptr, 0, 0} {}
+
+bool Base::hasSetDictionary() {
+  if (!has_set_dictionary_) {
```

Comment:
A little confuse here, has_set_dictionary_ is set to true, then hasSetDictionary return false?

Reply:
Hi @liuqiming-intel. Thanks for your review. `has_set_dictionary_` is a tag to record is dictionary has try to set in this context(should be by request). So if `has_set_dictionary_` is `false`, we will return false, and set it as `true`. So next round it will be true.